### ISSUE TITLE
Image Preview in Alacritty on macOS

### ISSUE THREAD

**Issue by yanzhang0219** [ranger/ranger#1839]
I am now using Alacritty on macOS and I cannot make the Image Preview feature work. This feature works on both my kitty and iTerm2. 

I installed w3m by `brew install w3m` successfully. Then I found that w3mimgdisplay was not included in this installation. I have done many investigations and research for this but I still don't know how to bring w3mimgdisplay to macOS. 

Please help! Thank you very much.

**Comment by kevinhwang91** [comment]
Ueberzug works fine with my alacritty, I think you should try it instead of w3m.

**Comment by yanzhang0219** [comment]
Thanks, but Ueberzug is unavailable on macOS.

**Comment by kevinhwang91** [comment]
Had you tried https://github.com/seebye/ueberzug/issues/61?

**Comment by yanzhang0219** [comment]
Yes. I installed it successfully before but when I use it as the preview method and tried to preview an image, I got this: 

![Screen Shot 2020-02-15 at 2 14 21 AM](https://user-images.githubusercontent.com/11582667/74586098-e5f6c700-4f98-11ea-80f3-e91bb5d72930.png)

So see, it doesn't work. Also, when I checked alacritty doc, it showed ueberzug is only for uses who run X11 in GNU/Linux. So macOS is not supported. 

BTW, my macOS is 10.15.3

Do you use ueberzug under macOS?

**Comment by kevinhwang91** [comment]
Oh sorry, I'm using Arch Linux.

**Comment by toonn** [comment]
Only preview method I'd kinda expect to work in alacritty on macOS is the mpv method.
The other alternatives are the native iTerm2 or Kitty previews.

**Comment by yanzhang0219** [comment]
Got it. I will try mpv. Thanks.

**Comment by yanzhang0219** [comment]
@toonn Hi, I just tried mpv method, but based on the wiki:
> Edit $ranger/core/fm.py then search for ImageDisplayer() and make the following changes

ImageDisplayer() does not exist in ranger/core/fm.py. Probably the source code has been changed. Then next, I used rg command to search for this piece of code but it cannot be found. 

Thanks.

**Comment by toonn** [comment]
It's in `ranger/ext/img_display.py`, has been for as long as I remember. I'd base  my patch on the mpv preview PR not the wiki, it's probably more recent.

**Comment by btwiusegentoo** [comment]
I'm trying to move from kitty. and as I see on the source code, you don't need to edit fm.py anymore.
but,  I think you have to do something with _@register_image_displayer_ on img_display.py?
i will try doing something on img_display.py.
I'm not sure what to do because i'm new to python or any programming language.
I'm trying to figure out now. the problem I have now is I get error No module named pathlib even through I did _pip3 install pathlib_. I think this problem I have maybe it has nothing to do with ranger lol maybe it's issue with my enviroment.
also, looks like other preview method uses def clean() and calls on quit or something?
so I think I have to do something with this cleanup() that it's unused.

**Comment by toonn** [comment]
@btwiusegentoo, if you're on linux the easiest preview method for alacritty should be ueberzug, unless you're running wayland.

Ranger shouldn't depend on anything outside of the standard library so I'm not sure where that error's coming from.

**Comment by btwiusegentoo** [comment]
@toonn sorry for confusing name. I went to Hackintosh from Gentoo now.
I was happy using Kitty on Wayland but it is laggy on mac so I'm trying Alacritty that looks snappier.
looks like that pathlib error was because mac uses python2 for default if specify only python on shebang.

**Comment by toonn** [comment]
Ranger works fine with python 2. And kitty doesn't lag at all on my ~10yo imac, just fyi.

**Comment by btwiusegentoo** [comment]
So probably my 2700X Ryzentosh that it's problematic?
or my Gentoo it was too fast because compiles all from source so it hasn't slow to launch kitty or slow to launch htop? lol
thanks, anyway I think I can't figure because I can't understand python so I will temporary go back to kitty. If I forgive launch time and that sometimes starts lagging to launch htop, I think it will be better for now.

**Comment by Boltzmachine** [comment]
> I am now using Alacritty on macOS and I cannot make the Image Preview feature work. This feature works on both my kitty and iTerm2.
> 
> I installed w3m by `brew install w3m` successfully. Then I found that w3mimgdisplay was not included in this installation. I have done many investigations and research for this but I still don't know how to bring w3mimgdisplay to macOS.
> 
> Please help! Thank you very much.

Same issue. I do not know how to display image in ranger since I do not use iTerm2 because it's too slow. I've tried compile w3m with ``--with-imagelib=imlib2`` as mentioned in [https://github.com/tats/w3m/issues/95](url). But it needs X11 library. So have you found a good way to achieve the image displaying?

**Comment by toonn** [comment]
The only preview methods I *know* work on macOS are iterm2 and kitty. Both require using a specific terminal.

**Comment by Boltzmachine** [comment]
> The only preview methods I _know_ work on macOS are iterm2 and kitty. Both require using a specific terminal.

Thanks. I'll try kitty.

**Comment by betaboon** [comment]
https://github.com/alacritty/alacritty/issues/910
https://github.com/alacritty/alacritty/pull/4763

**Comment by fwarmuth** [comment]
any news here?
